### PR TITLE
Add/abi decoding contracts models

### PR DESCRIPTION
**Pull request by austinFlipside**
- updates decoder macro to use gold logs
- adds `bronze_api__contract_abis` model + tests, with retry logic
- adds complete silver abi pipeline, with tests
- adds `dim_contract_abis`
- adds var logic to `bronze__decoded_logs_fr` for streamline v1
- removes streamline ABI model set up
- adds `dim_contracts` pipeline, including docs and tests
- will launch as `v3.1.0`

### CODE REVIEW

File: models/decoder_package/abis/bronze/_retry_abis.sql [hunk 1]
```diff
@@ -14,10 +14,14 @@ WITH retry AS (
...
         r
-        LEFT JOIN {{ ref("silver__verified_abis") }}
+        LEFT JOIN {{ source(
+            'silver',
+            'verified_abis'
+        ) }}
         v USING (contract_address)
     WHERE
-        r.total_interaction_count >= 1000 -- high interaction count
+        r.total_interaction_count >= 250
```
Comment (drethereum) [inline models/decoder_package/abis/bronze/_retry_abis.sql:23]: did you test if this was high enough to support all chains?

File: models/decoder_package/abis/bronze/_retry_abis.sql [hunk 2]
```diff
@@ -27,34 +31,45 @@ WITH retry AS (
...
+                ) }}
             WHERE
                 _inserted_timestamp >= CURRENT_DATE - INTERVAL '30 days' -- this won't let us retry the same contract within 30 days
+                AND abi_data:error is null
         )
     ORDER BY
         total_interaction_count DESC
     LIMIT
-        50
+        5
```
Comment (drethereum) [inline models/decoder_package/abis/bronze/_retry_abis.sql:45]: similar question to the above, is that high enough?

File: models/decoder_package/abis/bronze/bronze_api__contract_abis.sql [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    {{ log("BLOCK_EXPLORER_ABI_API_KEY_PATH: " ~ block_explorer_vault_path, info=True) }}
+
+{%- endif -%}
+
+{{ config(
+    materialized = 'incremental',
+    unique_key = "contract_address",
+    post_hook = "ALTER TABLE {{ this }} ADD SEARCH OPTIMIZATION on equality(contract_address)",
+    full_refresh = false,
+    tags = ['bronze_abis_core']
```
Comment (drethereum) [inline models/decoder_package/abis/bronze/bronze_api__contract_abis.sql:20]: this should be `bronze_abis` imo to follow the structure of the tags everywhere else - <schema>_<model_category>

File: models/decoder_package/abis/bronze/tests/test_bronze__contract_abis_recent.sql [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{{ config (
+    materialized = "view",
+    tags = ['daily_test']
```
Comment (drethereum) [inline models/decoder_package/abis/bronze/tests/test_bronze__contract_abis_recent.sql:3]: new tag eh? we would need to update the daily test workflows to account for this and update that tag on everything we want to be included in daily test if so. Currently we just have `dbt test --exclude full_test recent_test etc.`

File: models/decoder_package/abis/silver/silver__flat_event_abis.sql [hunk 5]
```diff
@@ -0,0 +1,117 @@
+{{ config (
+    materialized = 'incremental',
+    incremental_strategy = 'delete+insert',
+    unique_key = "contract_address",
+    cluster_by = "_inserted_timestamp::date",
+    tags = ['silver_abis'],
+    post_hook = "ALTER TABLE {{ this }} ADD SEARCH OPTIMIZATION ON EQUALITY (contract_address)"
```
Comment (drethereum) [inline models/decoder_package/abis/silver/silver__flat_event_abis.sql:7]: incredibly nitpicky i know...can you switch the order of `tags` and `post_hook` so tags are last?

File: models/decoder_package/abis/bronze/tests/test_bronze__contract_abis_recent.sql (hunk 4 above)
Comment (austinFlipside) [reply models/decoder_package/abis/bronze/tests/test_bronze__contract_abis_recent.sql:3]: There's a PR up in core models with these changes. We need another test job that runs once a day. We dont need to test recent ABI stuff multiple times a day, for example. Full test is intended to be monthly

Comment (austinFlipside) [reply models/decoder_package/abis/bronze/tests/test_bronze__contract_abis_recent.sql:3]: - intraday: looking at chainhead and core models several times a day
- daily: checking recent curated data, abis, etc
- full_test: everything, monthly

**Review by drethereum — APPROVED** [review]

**Pull request merged**